src/_lib/modules/word-cloud/word-cloud.component.spec.ts

```diff
@@ -0,0 +1,162 @@
+            comp['_positionedWords'] = [
+                { x: 20, y: 20, width: 10, height: 10 },
+                { x: 30, y: 30, width: 10, height: 10 },
+                { x: 40, y: 40, width: 10, height: 10 },
+            ] as IWordWithPosition<any>;
+            const result = comp['_isIntersecting'](boundingBox);
+            expect(result).toBe(false);
+        });
+
+        it('should return true if bounding box overlaps with at least one other bounding box', () => {
+            const boundingBox: IBoundingBox = { x: 10, y: 10, width: 20, height: 20 };
+            comp['_positionedWords'] = [
+                { x: 0, y: 0, width: 10, height: 10 },
+                { x: 30, y: 30, width: 10, height: 10 },
+                { x: 10, y: 15, width: 10, height: 10 },
+            ] as IWordWithPosition<any>;
+            const result = comp['_isIntersecting'](boundingBox);
+            expect(result).toBe(true);
+        });
+
+        it('should return true if bounding box completely overlaps with another bounding box', () => {
+            const boundingBox: IBoundingBox = { x: 10, y: 10, width: 20, height: 20 };
+            comp['_positionedWords'] = [
+                { x: 0, y: 0, width: 50, height: 50 },
+                { x: 30, y: 30, width: 10, height: 10 },
+            ] as IWordWithPosition<any>;
+            const result = comp['_isIntersecting'](boundingBox);
+            expect(result).toBe(true);
+        });
+
+        it('should return true if bounding box is completely inside another bounding box', () => {
+            const boundingBox: IBoundingBox = { x: 10, y: 10, width: 5, height: 5 };
+            comp['_positionedWords'] = [
+                { x: 0, y: 0, width: 50, height: 50 },
+                { x: 30, y: 30, width: 10, height: 10 },
+            ] as IWordWithPosition<any>;
+            const result = comp['_isIntersecting'](boundingBox);
+            expect(result).toBe(true);
+        });
+
+        it('should return true if bounding box is partially inside another bounding box', () => {
+            const boundingBox: IBoundingBox = { x: 10, y: 10, width: 20, height: 20 };
+            comp['_positionedWords'] = [
+                { x: 0, y: 0, width: 50, height: 50 },
+                { x: 30, y: 30, width: 10, height: 10 },
+                { x: 5, y: 5, width: 20, height: 20 },
+            ] as IWordWithPosition<any>;
+            const result = comp['_isIntersecting'](boundingBox);
+            expect(result).toBe(true);
+        });
+    });
+
+    describe('_getTruncatedWordsWithFontSize', () => {
+        beforeEach(() => {
+            comp.options = { maxCharCount: 20 };
+            comp.words = [];
+            comp.ngOnChanges({
+                options: new SimpleChange(undefined, comp.options, true)
+            });
+        });
+
+        it('should not truncate words that do not exceed the max character count', () => {
+            const word = cloneDeep(mockWord);
```

Comment:
mockWord is only one level deep.   Do we need to use `cloneDeep` instead of `const word = {...mockWord}`?

Reply:
You're right, no need for cloneDeep. I'll update